nncf/common/graph/graph.py

```diff
@@ -252,6 +252,24 @@ def get_all_nodes(self) -> List[NNCFNode]:
             all_nodes.append(nncf_node)
         return all_nodes
 
+    def get_all_simple_paths(self,
```

Comment:
It looks like this is a separate algorithm as in [NetworkX](https://networkx.org/documentation/networkx-1.9/index.html). What do you think about moving this algorithm into a separate module?

Reply:
This method just a wrapper for method from networkx: `nx.all_simple_paths`
I don't think it makes sense to move it to a separate module, because it heavily relies on fields of `NNCFGraph` - 5 entries of `self.`.